flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/concurrent/RunningInMainThreadAssert.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.concurrent;
+
+/**
+ * {@code RunningInMainThreadAssert} provides a method for verifying that the current thread is the
+ * main thread.
+ */
+@FunctionalInterface
+public interface RunningInMainThreadAssert {
```

Comment:
I thought it would be helpful to have a simple function interface in cases like the `DefaultResourceCleaner` where we're not utilizing the `ComponentMainThreadExecutor` but just want to check whether the code is running in the mainthread.

Reply:
Let's remove it, it seems bit off to have this as a separate interface (is only intended for the ComponentMainThreadExecutor).

Reply:
ok, already done 👍